### DISCUSSION TITLE
Boolean Accuracy

### DISCUSSION THREAD

**Discussion by z3dev** [jscad/OpenJSCAD.org#1020] [General]
So, I've been playing around with some of the 2D boolean libraries in order to study the algorithms, and access if useful inside JSCAD. A lot of these look promising but every library has issues, and most maintenance is slow or none.

What I did find was a nice tool put together by @daef call [poly-bool-comparison](https://github.com/daef/poly-bool-comparison). It's really well done and shows a summation of the issues in each library. Here's the viewer

https://daef.github.io/poly-bool-comparison/dissection/

Also, I have taken the samples in this tool, and created a design to test out JSCAD 2D booleans. JSCAD is pretty impressive.

**Comment by z3dev** [comment]
[poly-bool-comparison.zip](https://github.com/jscad/OpenJSCAD.org/files/8189886/poly-bool-comparison.zip)

**Comment by daef** [comment]
if you create a run including jscad results I'd be happy to merge them :)

**Comment by platypii** [comment]
I made a fork of `poly-bool-comparison` which adds JSCAD V2 to the comparison. I will eventually make a PR to the comparison repo, but I wanted to share here first.

This is run on JSCAD V2. Many of the outputs are missing because the geometry contain gaps and cannot be converted to outlines for conversion to SVG.

V3 will be a **significant** improvement.

https://github.com/platypii/poly-bool-comparison/tree/jscad

![jscad-union](https://user-images.githubusercontent.com/1766297/211759256-bfd41072-5ca0-458c-9217-40420d7eeae4.png)

**Comment by z3dev** [comment]
@daef i have some improvements as well. Do you still want changes?

**Reply by daef** [reply]
sorry, I'm quite busy. but sure, I'm always open to improvements.
